replacer_test.go

```diff
@@ -59,6 +59,26 @@ func TestReplacer(t *testing.T) {
 			input:  `{"json": "object"}`,
 			expect: "",
 		},
+		{
+			input:  `\{"json": "object"}`,
+			expect: `{"json": "object"}`,
+		},
+		{
+			input:  `\{"json": "object"\}`,
+			expect: `{"json": "object"}`,
+		},
+		{
+			input:  `\{"json": "object{bar}"\}`,
+			expect: `{"json": "object"}`,
+		},
+		{
+			input:  `\{"json": \{"nested": "object"\}\}`,
+			expect: `{"json": {"nested": "object"}}`,
+		},
+		{
+			input:  `\{"json": \{"nested": "{bar}"\}\}`,
+			expect: `{"json": {"nested": ""}}`,
+		},
```

Comment:
Can we add more test cases? Specifically:

- backslashes that aren't acting as escapes
- malformed escapes (if there is such a thing)
- escaping only the closing brace
- repeated backslashes
- content before and after escaped brace segments

Or any others you can think of, but I think this would be pretty comprehensive.

Reply:
I've added additional test cases, along with one that prompted issue #3116 and these all pass.

Reply:
Awesome, they LGTM!